clarity/contracts/arkadiko-stake-pool-diko-v1-1.clar

```diff
@@ -7,320 +7,137 @@
 
 ;; Constants
 (define-constant POOL-TOKEN .arkadiko-token)
-(define-constant BLOCKS-PER-YEAR u52560)
 
 ;; Variables
-(define-data-var token-uri (string-utf8 256) u"")
-(define-data-var total-staked uint u0)
-(define-data-var cumm-reward-per-stake uint u0)
-(define-data-var last-reward-increase-block uint u0) 
-
+(define-data-var last-reward-add-block uint u0)
 
 ;; ---------------------------------------------------------
-;; SIP-10 Functions
+;; Stake Functions
 ;; ---------------------------------------------------------
 
-(define-fungible-token stdiko)
-
-(define-read-only (get-total-supply)
-  (ok (ft-get-supply stdiko))
-)
-
-(define-read-only (get-name)
-  (ok "Staked Arkadiko Token")
-)
 
-(define-read-only (get-symbol)
-  (ok "stDIKO")
+;; Get variable last-reward-add-block
+(define-read-only (get-last-reward-add-block)
+  (var-get last-reward-add-block)
 )
 
-(define-read-only (get-decimals)
-  (ok u6)
-)
-
-(define-read-only (get-balance (account principal))
-  (ok (ft-get-balance stdiko account))
-)
+;; DIKO (staked & rewards) over total supply of stDIKO - Result with 6 decimals
+(define-read-only (diko-stdiko-ratio)
+  (let (
+    ;; Total stDIKO supply
+    (stdiko-supply (unwrap-panic (contract-call? .stdiko-token get-total-supply)))
 
-(define-public (set-token-uri (value (string-utf8 256)))
-  (if (is-eq tx-sender (contract-call? .arkadiko-dao get-dao-owner))
-    (ok (var-set token-uri value))
-    (err ERR-NOT-AUTHORIZED)
+    ;; Total DIKO (staked + rewards)
+    (diko-supply (unwrap-panic (contract-call? .arkadiko-token get-balance (as-contract tx-sender))))
   )
-)
-
-(define-read-only (get-token-uri)
-  (ok (some (var-get token-uri)))
-)
-
-(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
-  (match (ft-transfer? stdiko amount sender recipient)
-    response (begin
-      (print memo)
-      (ok response)
+    (if (is-eq stdiko-supply u0)
+      u1000000
+      (/ (* diko-supply u1000000) stdiko-supply)
     )
-    error (err error)
   )
 )
 
-;; ---------------------------------------------------------
-;; Stake Functions
-;; ---------------------------------------------------------
-
-;; Keep track of total amount staked and last cumm reward per stake
-(define-map stakes 
-   { staker: principal } 
-   {
-      uamount: uint,  ;; micro diko amount staked
-      cumm-reward-per-stake: uint
-   }
-)
-
-(define-read-only (get-stake-of (staker principal))
-  (default-to
-    { uamount: u0, cumm-reward-per-stake: u0 }
-    (map-get? stakes { staker: staker })
-  )
-)
-
-;; Get stake info - amount staked
-(define-read-only (get-stake-amount-of (staker principal))
-  (get uamount (get-stake-of staker))
-)
-
-;; Get stake info - last rewards block
-(define-read-only (get-stake-cumm-reward-per-stake-of (staker principal))
-  (get cumm-reward-per-stake (get-stake-of staker))
-)
-
-;; Get variable total-staked
-(define-read-only (get-total-staked)
```

Comment:
@nieldeckx this was removed and we call this in the front-end. Should we implement this method on `arkadiko-stake-pool-diko-v1-1` and just call the FT supply function of stDIKO?

Reply:
@philipdesmedt I will again add this method and calculate the amount of DIKO you would get if you unstaked with all the stDIKO from your wallet.

As DIKO is now auto-compounding you will see the 'staked value' increasing over time. This also means there is no need to show 'pending rewards' on this pool.

Reply:
OK thanks! I already added the method but feel free to adjust/fix

Reply:
Fixed it. The method will now return the amount of DIKO you will get when unstaking - so including compounded rewards.

Reply:
Great, thx!